library/components/tableComponents/TSelectColumns.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+  <t-dropdown
+    :disable-handle-class="isExpanded"
+    :is-active="checked.length ? true : false"
+  >
+    <!-- Handle -->
+    <div
+      slot="handle"
+      class="flex items-center gap-1 p-1 text-sm font-medium"
+      v-if="!isExpanded"
+    >
+      <view-column-outline size="20" />
+      <span> Modify Columns </span>
+      <menu-down-icon size="20" />
+    </div>
+
+    <!-- Popup Contents -->
+    <div class="min-w-[294px]" attrs.slot="outside">
+      <div
+        class="
+          px-[12px]
+          pt-[16px]
+          pb-[8px]
+          flex
+          justify-between
+          items-center
+          w-full
+        "
+      >
+        <h6
+          class="
+            text-[10px] text-[#727184]
+            font-semibold
+            uppercase
+            leading-[15px]
+            tracking-widest
+          "
+        >
+          Columns
+        </h6>
+        <span
+          class="
+            text-[#145DEB] text-[13px]
+            cursor-pointer
+            font-normal
+            leading-[18px]
+          "
+          :class="checked.length ? 'text-[#145DEB]' : 'text-[#727184]'"
+        >
+          <span
+            v-if="checked.length < filterableColumns.length"
+            @click="selectAll"
+          >
+            Select All
+          </span>
+          <span v-else @click="deselectAll"> Reset </span>
+        </span>
+      </div>
+      <div class="px-[8px] pt-[4px] pb-[6px] w-full">
+        <ul class="max-h-[320px] overflow-auto">
+          <li
+            class="
+              flex
+              justify-between
+              px-[8px]
+              pb-[1px]
+              text-sm
+              cursor-pointer
+              text-[#555463]
+            "
+            v-for="(item, index) in internalColumns"
+            :key="index"
+          >
+            <div
+              class="
+                flex
+                items-center
+                justify-between
+                w-full
+                hover:text-[#1C1C21]
+                leading-[16.41px]
+              "
+            >
+              <base-checkbox
+                class="!min-w-[200px]"
+                :label="item.label"
+                :value="item"
+                v-model="checked"
+              />
+            </div>
+          </li>
+          <small v-if="!filterableColumns.length"> No items found </small>
+        </ul>
+      </div>
+    </div>
+  </t-dropdown>
```

Comment:
Was t-multi-selector not usable for this use case?

Reply:
I could have used t-multi-selector, but only by adding a ton of special-case code to it that would have made it much harder to reason about and edit in the future. t-mulit-selector assumes the data to display is coming from a layer, but given that the data for this makes more sense to provide as a prop it was not a great fit under the hood. 😢

Reply:
ah, ok. Understood.